### PR TITLE
Batch verification

### LINKED ISSUE #1177: Integrate Batch Verification module in Crypography module

### PR DESCRIPTION
**Pull request by franssl**
### New Pull Request Submissions:

1. [ ] Have you followed the guidelines in our [Contributing document](https://github.com/atlascity/Community/tree/master/CONTRIBUTING.md)?
2. [ ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
3. [ ] I have added tests to cover my changes.
4. [ ] All new and existing tests passed.
5. [ ] Have you lint your code locally prior to submission?
6. [ ] Does your code follows the code style of this project?
7. [ ] Does your change require a change to the documentation.
    - [ ] I have updated the documentation accordingly.
9. [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
10. [ ] Have you inserted a keyword and link to the issues the PR closes in its descriptions (ex `closes #1`) ?
11. [ ] Is you branch up to date, have you integrated all the latest changes from develop and resolved conflicts ?

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:
Closes #1177

### CODE REVIEW

File: src/Catalyst.Core.Lib.Tests/UnitTests/Validators/TransactionValidatorTests.cs [hunk 1]
```diff
@@ -166,6 +167,8 @@ public bool Verify(ISignature signature, ReadOnlySpan<byte> message, ReadOnlySpa
             {
                 return _verifyResult;
             }
+
+            public bool BatchVerify(IList<ISignature> signatures, IList<byte[]> messages, ReadOnlySpan<byte> context) { throw new NotImplementedException(); }
```
Comment (sudhirtibrewal) [inline src/Catalyst.Core.Lib.Tests/UnitTests/Validators/TransactionValidatorTests.cs]: is this correct?

File: src/Catalyst.Benchmark/Catalyst.Abstractions/CryptoBenchmark.cs [hunk 2]
```diff
@@ -95,6 +96,8 @@ public bool Verify(ISignature signature, ReadOnlySpan<byte> message, ReadOnlySpa
                 return true;
             }
 
+            public bool BatchVerify(IList<ISignature> signatures, IList<byte[]> messages, ReadOnlySpan<byte> context) { throw new NotImplementedException(); }
```
Comment (sudhirtibrewal) [inline src/Catalyst.Benchmark/Catalyst.Abstractions/CryptoBenchmark.cs]: is this correct?

File: src/Catalyst.Core.Modules.KeySigner.Tests/UnitTests/KeySignerTests.cs [hunk 3]
```diff
@@ -148,6 +149,7 @@ private sealed class CryptoContext : ICryptoContext
 
             public ISignature Sign(IPrivateKey privateKey, ReadOnlySpan<byte> message, ReadOnlySpan<byte> context) => _signature;
             public bool Verify(ISignature signature, ReadOnlySpan<byte> message, ReadOnlySpan<byte> context) { throw new NotImplementedException(); }
+            public bool BatchVerify(IList<ISignature> signatures, IList<byte[]> messages, ReadOnlySpan<byte> context) { throw new NotImplementedException(); }
```
Comment (sudhirtibrewal) [inline src/Catalyst.Core.Modules.KeySigner.Tests/UnitTests/KeySignerTests.cs]: are test ignored?

Comment (franssl) [reply src/Catalyst.Core.Modules.KeySigner.Tests/UnitTests/KeySignerTests.cs]: Yeah we don't need an implementation of batch verification in the KeySigner tests

File: src/Catalyst.Core.Lib.Tests/UnitTests/Validators/TransactionValidatorTests.cs (hunk 1 above)
Comment (franssl) [reply src/Catalyst.Core.Lib.Tests/UnitTests/Validators/TransactionValidatorTests.cs]: Same as below, batch verification isn't needed for these tests

File: src/Catalyst.Benchmark/Catalyst.Abstractions/CryptoBenchmark.cs (hunk 2 above)
Comment (franssl) [reply src/Catalyst.Benchmark/Catalyst.Abstractions/CryptoBenchmark.cs]: There is actually another repo I made a while ago with benchmarks for our crypto, including batch verification. I didn't know this project here existed until doing this bit of work. I'll make a ticket for consolidating the benchmarking into one place after testnet.

Comment (franssl) [reply src/Catalyst.Benchmark/Catalyst.Abstractions/CryptoBenchmark.cs]: I've created #1199 to fix this

**Pull request merged**